serverless-graphql-api/javascript-appsync-sam-js/README.md

```diff
@@ -0,0 +1,144 @@
+```
+
+## Manually deploy the sample application
+***Note:** Before deploying application manually first time you will need to deploy shared Amazon Cognito stack, see previous section for details.*
+
+To build and deploy your application for the first time, run the following in your shell:
+
+```bash
+npm install
+sam build
+sam deploy --guided --stack-name javascript-appsync-sam
+```
+
+The first command will build the source of your application. The second command will package and deploy your application to AWS, with a series of prompts:
+
+* **Stack Name**: The name of the stack to deploy to AWS CloudFormation. This should be unique to your account and region, and a good starting point would be something matching your project name.
+* **AWS Region**: The AWS region you want to deploy your app to.
+* **Parameter CognitoStackName**: The shared Amazon Cognito stack name 
+* **Confirm changes before deploy**: If set to yes, any change sets will be shown to you before execution for manual review. If set to no, the AWS SAM CLI will automatically deploy application changes.
+* **Allow AWS SAM CLI IAM role creation**: Many AWS SAM templates, including this example, create AWS IAM roles required for the AWS Lambda function(s) included to access AWS services. By default, these are scoped down to minimum required permissions. To deploy an CloudFormation stack which creates or modifies IAM roles, the `CAPABILITY_IAM` value for `capabilities` must be provided. If permission isn't provided through this prompt, to deploy this example you must explicitly pass `--capabilities CAPABILITY_IAM` to the `sam deploy` command.
+* **Save arguments to samconfig.toml**: If set to yes, your choices will be saved to a configuration file inside the project, so that in the future you can just re-run `sam deploy` without parameters to deploy changes to your application.
+
+The AWS AppSync endpoint API will be displayed in the outputs when the deployment is complete.
+
+## Testing
+Unit tests are defined in the `__tests__\unit` folder in this project. Integration tests are defined in the `__tests__\integration` folder.
+
+Use `npm` to install the [Jest test framework](https://jestjs.io/) and run the tests:
+
+```bash
+my-application$ npm install
+```
+
+Unit tests use AWS SDK to evaluate AWS AppSync resolver code. They use mock data, not the integrations with the backend services. Use following command to run unit tests:
+```bash
+npm run test:unit
+```
+
+Integration tests send GraphQL queries to the AWS AppSync endpoint and verify responses received. In addition, integration tests use WebSocket connection to the AWS AppSync for GraphQL subscriptions. You should run the test initialization script first, to create necessary accounts in Amazon Cognito and retrieve access tokens. Also make sure to cleanup after integration testing is over:
+
+```bash
+eval "$(./__tests__/integration/test_init.sh)" 
```

Comment:
I get an error the first time I run this. Is that expected?

```
An error occurred (UserNotFoundException) when calling the AdminDeleteUser operation: User does not exist.
```

Reply:
Fixed